### PR TITLE
Articles

### PR DESCRIPTION
**Pull request by anastasiiatulentseva**
add articles management features

### CODE REVIEW

File: app/assets/javascripts/articles.coffee [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+do ->
+  onLoad = ->
+    window['rangy'].initilized = false
+    $('.text-wysiwyg').wysihtml5({'toolbar': {'html': true}})
+
+  $(document).on('turbolinks:load', onLoad)
```
Comment (stulentsev) [inline app/assets/javascripts/articles.coffee]: Last line must end with a newline, as per POSIX conventions. Set up your editor. :)

File: app/assets/javascripts/articles.coffee [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+do ->
+  onLoad = ->
```
Comment (stulentsev) [inline app/assets/javascripts/articles.coffee]: Now that we don't have to connect this handler to two different places, you can just inline it below.

File: app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @article = Article.find(params[:id]).destroy
+    flash[:info] = "Article deleted"
+    redirect_to articles_path
+  end
+  
+  private
+  
+  def article_params
+    params.require(:article).permit(:name, :text, :category, :order)
+  end
```
Comment (stulentsev) [inline app/controllers/articles_controller.rb:51]: No authorization here?

File: app/views/articles/_article_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  <div class="form-group">
+    <%= f.label :order, class: "control-label" %>
+    <%= f.text_field :order, class: "form-control" %>
+  </div>
+  <div class="btn-group">
+    <%= f.submit "Save", class: "btn btn-primary" %>
+    <%= link_to "Delete", @article, method: :delete, data: { confirm: "Delete article?" }, class: "btn btn-danger"%>
+  </div>
+<% end %>
+
```
Comment (stulentsev) [inline app/views/articles/_article_form.html.erb]: Extra spaces.

File: app/views/articles/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<% provide(:title, "Adding new article") %>
+
+<h1 class="page-header">New article</h1>
+
+<div class="col-md-9 col-md-offset-2">
+  <%= render 'articles/article_form' %>
+</div>
```
Comment (stulentsev) [inline app/views/articles/new.html.erb]: newline

File: app/views/articles/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<%= provide(:title, @article.name) %>
+
+<h1 class="page-header"><%= @article.name %></h1>
+<div class="pull-right">
+  <%= link_to 'Edit article', edit_article_path(@article.id), class: "text-danger" %> | 
+  <%= link_to 'Back to index', articles_path, class: "text-danger" %>
+</div>
+<div class="clearfix"><%= @article.text.html_safe %></div>
```
Comment (stulentsev) [inline app/views/articles/show.html.erb]: You might want to learn about HAML. Simplifies the markup greatly. This whole file, for example, would look like this:

``` haml
= provide(:title, @article.name)
%h1.page-header= @article.name
.pull-right
  = link_to 'Edit article', edit_article_path(@article.id), class: "text-danger"
  |
  = link_to 'Back to index', articles_path, class: "text-danger"
.clearfix= @article.text.html_safe
```

File: app/controllers/articles_controller.rb (hunk 3 above)
Comment (anastasiiatulentseva) [inline app/controllers/articles_controller.rb:51]: not yet.

File: app/assets/javascripts/articles.coffee (hunk 2 above)
Comment (anastasiiatulentseva) [inline app/assets/javascripts/articles.coffee]: done

File: app/views/articles/_article_form.html.erb (hunk 4 above)
Comment (anastasiiatulentseva) [inline app/views/articles/_article_form.html.erb]: done

File: app/views/articles/new.html.erb (hunk 5 above)
Comment (anastasiiatulentseva) [inline app/views/articles/new.html.erb]: done

File: app/assets/javascripts/articles.coffee (hunk 1 above)
Comment (anastasiiatulentseva) [inline app/assets/javascripts/articles.coffee]: done

File: app/views/articles/show.html.erb (hunk 6 above)
Comment (anastasiiatulentseva) [inline app/views/articles/show.html.erb]: ok!

File: app/views/articles/_article.html.haml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+%li
+  = link_to article.name, article_path(article.id)
```
Comment (stulentsev) [inline app/views/articles/_article.html.haml]: newline :)

File: app/views/articles/_article_form.html.haml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    = f.label :name, class: "control-label"
+    = f.text_field :name, class: "form-control", placeholder: "Enter artilcle name"
+  .form-group
+    = f.label :text, class: "control-label"
+    = f.text_area :text, class: "text-wysiwyg form-control", rows: 15
+  .form-group
+    = f.label :category, class: "control-label"
+    = f.text_field :category, list: 'category', class: "form-control"
+    %datalist#category
+      - Article.select(:category).distinct.each do |article|
```
Comment (stulentsev) [inline app/views/articles/_article_form.html.haml]: No queries in views. Make it a controller var.

File: app/views/articles/_article_form.html.haml [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        %option{:value => "#{article.category}"}
```
Comment (stulentsev) [inline app/views/articles/_article_form.html.haml]: can be just

```
%option{value: article.category}
```

File: app/views/articles/_article_form.html.haml (hunk 8 above)
Comment (anastasiiatulentseva) [inline app/views/articles/_article_form.html.haml]: done

File: app/views/articles/_article.html.haml (hunk 7 above)
Comment (anastasiiatulentseva) [inline app/views/articles/_article.html.haml]: not here )

File: app/views/articles/_article_form.html.haml (hunk 9 above)
Comment (anastasiiatulentseva) [inline app/views/articles/_article_form.html.haml]: done

**Pull request merged**